src/coreclr/nativeaot/Runtime/gcinfodecoder.h

```diff
@@ -1,4 +1,7 @@
 // Licensed to the .NET Foundation under one or more agreements.
 // The .NET Foundation licenses this file to you under the MIT license.
 
+// USE_GC_INFO_DECODER does not defined for Win-x86 in CoreCLR
```

Comment:
This won't be enough. You will also need to define it when building RyuJIT.

Reply:
I assume this is NativeAOT specific customization to https://github.com/dotnet/runtimelab/blob/feature/NativeAOT/src/coreclr/jit/CMakeLists.txt

Also I did not see any reference to `USE_GC_INFO_DECODER` in jit folder. Are you talking that I need to use `FEATURE_EH_FUNCLETS` then how I make that change purely NativeAOT specific ?

Reply:
> I assume this is NativeAOT specific customization to https://github.com/dotnet/runtimelab/blob/feature/NativeAOT/src/coreclr/jit/CMakeLists.txt

Yes. It becomes tricky because it would mean that the JIT binary used by CoreCLR will be different from the JIT binary used by AOT. It is actually a bunch of binaries since we build multiple variants to support cross-compilation. There are short-term hacks possible in this branch. Once/if we move NativeAOT to dotnet/runtime, we will need to be able to use the same JIT binary for both regular CoreCLR and NativeAOT. If we keep the ABI differences, we may need to replace the ifdefs with dynamic checks in the JIT. Other option is to just stick with the same ABI as CoreCLR and clean it up as necessary (ie avoid forking the ABI for NativeAOT).

> Also I did not see any reference to USE_GC_INFO_DECODER in jit folder

JIT has `JIT32_GCENCODER`. `JIT32_GCENCODER = !USE_GC_INFO_DECODER`

Reply:
I think I found a way how we can postpone decision temporary. Instead of this particular change, I remove these lines from .h files

https://github.com/dotnet/runtimelab/blob/6f2cfcdb8a4484f5a62234353923ee8073c57593/src/coreclr/jit/jit.h#L468-L470

and 

https://github.com/dotnet/runtimelab/blob/6f2cfcdb8a4484f5a62234353923ee8073c57593/src/coreclr/inc/gcinfodecoder.h#L22-L24

And will control these defines from CMake files. That change maybe be submitted to Runtime repo I think, and all hacks would be in CMake files in NativeAOT repo. That way we can explore all 3 options without too much commitment to either of them, if I only apply CMake hacks locally. What do you think?

Reply:
I do not think that there is a big difference in controlling it in cmake files vs. in header files. Both of these are short-term hacks that would be ok in this branch, but not in dotnet/runtime. You end up with a JIT build that is different from regular CoreCLR JIT build in either case.

Reply:
Ok, will try that. I think about JIT as black box for now, so it was a scary change for me. I think I understand direction now.